### PR TITLE
Allow toggle for 5xx response content from endpoints #patch

### PR DESCRIPTION
**Pull request by raghavtan**
Currently the sidecar doesn't create file (from endpoints giving 5xx response) on the volume mounted on the pod 

The new `bool` environment variable `ENABLE_5XX` 
- enable an option/toggle to save/write content from 5xx upstream/endpoint response

Use Case: dynamically serving custom error pages from ambassador via endpoints 

#patch

### CODE REVIEW

**Review by jekkel — CHANGES_REQUESTED** [review]
Thanks for your contribution! PR looks good already, some minor things left.

File: README.md [hunk 1]
```diff
@@ -171,3 +171,8 @@ If the filename ends with `.url` suffix, the content will be processed as an URL
   - description: if this is given and points to a file or `~/.kube/config` is mounted k8s config will be loaded from this file, otherwise "incluster" k8s configuration is tried.
   - required: false
   - type: string
+
+- `ENABLE_5XX`
+  - description: Set to true to enable pulling of 5XX response content from config map
```
[the remark this answers is not in the document]
Comment (raghavtan) [reply README.md]: @jekkel Updated readme documentation to reference `*.url` feature for `ENABLE_5XX` env var

File: sidecar/sidecar.py [hunk 2]
```diff
@@ -64,14 +65,23 @@ def main():
...
+    else:
+        print(f"{timestamp()} 5xx response content will not be enabled.")
+        enable_5xx = False
+
     current_namespace = open("/var/run/secrets/kubernetes.io/serviceaccount/namespace").read()
     if os.getenv(METHOD) == "LIST":
         for res in resources:
             list_resources(label, label_value, target_folder, url, method, payload,
-                           current_namespace, folder_annotation, res, unique_filenames, script)
+                           current_namespace, folder_annotation, res, unique_filenames, script, enable_5xx)
```
[the remark this answers is not in the document]
Comment (raghavtan) [reply sidecar/sidecar.py:80]: @jekkel refactored the order of args for method `list_resources` definition

File: sidecar/helpers.py [hunk 3]
```diff
@@ -88,11 +88,12 @@ def request(url, method, payload=None):
         auth = None
 
     r = requests.Session()
+
     retries = Retry(total=retry_total,
                     connect=retry_connect,
                     read=retry_read,
                     backoff_factor=retry_backoff_factor,
-                    status_forcelist=[500, 502, 503, 504])
+                    status_forcelist=list() if enable_5xx else [500, 502, 503, 504])
```
[the remark this answers is not in the document]
Comment (raghavtan) [reply sidecar/helpers.py]: added new variable `enforce_status_codes` to replace inline expression

**Review by jekkel — COMMENTED** [review]
All changes look good so far. Can imagine adding a test to circleCI test script verifying the url feature works as intended with and without 500 responses? Ideally this should not do any requests to the outside world, we probably should install a small static webserver to the kind k8s cluster serving some files...? WDYT?

**Comment by raghavtan** [comment]
> All changes look good so far. Can you imagine adding a test to circleCI test script verifying the url feature works as intended with and without 500 responses? Ideally this should not do any requests to the outside world, we probably should install a small static webserver to the kind k8s cluster serving some files...? WDYT?

This wound aptly be 2x all the tests with and without `ENABLE_5XX` flag
- 2 separate sidecar pods with different flags 
- a dummy http server pod serving custom content html pages which can be compared with `verify_resources_read` 

Adding a sample test scenario with multiple sidecar containers and validation to `verify_resources_read`  (not adding the dummy static server for now)

@jekkel how would this setup be in your opinion ?

**Comment by jekkel** [comment]
Yes, that's exactly the kind of setup I was thinking about. Would you be able to integrate that into this PR?

**Comment by raghavtan** [comment]
> Yes, that's exactly the kind of setup I was thinking about. Would you be able to integrate that into this PR?

Added @jekkel 
- test scenario script updated
- deployment for dummy server image
- added dummy server

**Review by jekkel — APPROVED** [review]

**Comment by jekkel** [comment]
We need to fix a CI issue but the feature is going to be in the next version. Thanks for your contribution again!

**Comment by raghavtan** [comment]
> We need to fix a CI issue but the feature is going to be in the next version. Thanks for your contribution again!

@jekkel not to pester, but when can we expect the CI to be fixed and the feature released. Need the feature to implement it on our edge-stack

**Comment by jekkel** [comment]
It's a matter of hours or days at most, sorry for any inconvenience.

**Comment by raghavtan** [comment]
No worries at all, lemme know if i can be any help in the process

**Comment by jekkel** [comment]
Ping @manu11th

**Comment by jekkel** [comment]
@raghavtan FYI released [v1.12.1](https://github.com/kiwigrid/k8s-sidecar/releases/tag/1.12.1)

**Pull request merged**